presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/CachingJdbcClient.java

```diff
@@ -106,8 +108,8 @@ public boolean schemaExists(ConnectorSession session, String schema)
         if (tableHandle.getColumns().isPresent()) {
             return tableHandle.getColumns().get();
         }
-        ColumnsCacheKey key = new ColumnsCacheKey(JdbcIdentity.from(session), tableHandle.getSchemaTableName());
-        return get(columnsCache, key, () -> delegate.getColumns(session, tableHandle));
+
+        return get(columnsCache, ColumnsCacheKey.forSessionAndTable(session, tableHandle.getSchemaTableName()), () -> delegate.getColumns(session, tableHandle));
```

Comment:
It has to be done for all caches, not only columns.

Reply:
> It has to be done for all caches, not only columns.

@kokosing do you have a usecase in mind?